distributed/client.py

```diff
@@ -1998,7 +2004,37 @@ def f(worker=None):
                 return None
 
         workers = sync(self.loop, self._run, f)
-        return {'scheduler': scheduler, 'workers': workers, 'client': client}
+        result = {'scheduler': scheduler, 'workers': workers, 'client': client}
+
+        if check:
+
+            # we care about the required & optional packages matching
+            extract = lambda x: merge(result[x]['packages'].values())
+            client_versions = extract('client')
+            scheduler_versions = extract('scheduler')
+
+            for pkg, cv in client_versions.items():
+                sv = scheduler_versions[pkg]
+                if sv != cv:
+                    raise ValueError("package [{package}] is version [{client}] "
+                                     "on client and [{scheduler}] on scheduler!".format(
+                                         package=pkg,
+                                         client=cv,
+                                         scheduler=sv))
+
+            for wkr, worker in workers.items():
```

Comment:
We often use the pronouns `w` or `worker` to mean a worker address, like `127.0.0.1:8786`.  Recommend changing `wkr` to w or worker and changing worker to some other pronoun like `response` or `d`